### PR TITLE
finished assignments

### PR DESCRIPTION
**Pull request by codecat226**

### CODE REVIEW

File: src/assignments/2-operator.js [hunk 1]
```diff
@@ -14,6 +14,25 @@ output example
...
+const add = num1 + num2;
+console.log(`${num1}+${num2}=${add}`);
+
+const sub = num1 - num2;
+console.log(`${num1}-${num2}=${sub}`);
+
+const multiply = num1 * num2;
+console.log(`${num1}*${num2}=${multiply}`);
+
+const divide = (num1 / num2).toFixed(2);
```
Comment (thoanguyen2907) [inline src/assignments/2-operator.js]: actually, in this case , it is quite necessary to check whether the num2 > 0 or not, then do the calculation.

File: src/assignments/2-operator.js [hunk 2]
```diff
@@ -14,6 +14,25 @@ output example
...
+console.log(`${num1}/${num2}=${divide}`);
+
+const remain = num1 % num2;
+console.log(`${num1}%${num2}=${remain}`);
```
Comment (thoanguyen2907) [inline src/assignments/2-operator.js]: here is also 
 in this case , it is quite necessary to check whether the num2 > 0 or not, then do the calculation.

File: src/assignments/4-switch.js [hunk 3]
```diff
@@ -8,3 +8,37 @@ vowel
...
+			break;
+		case 'u':
+			console.log("vowel");
+			break;
+		case 'a':
+			console.log("vowel");
+			break;
+		default:
+			console.log("consonant");
+	}
```
Comment (thoanguyen2907) [inline src/assignments/4-switch.js:42]: good for switch case 👍

File: src/assignments/6-array.js [hunk 4]
```diff
@@ -21,3 +23,17 @@ min=1
...
+let min = input[0]; //1
+
+for (let i = 1; i < input.length; i++) {
+	if (input[i] < min) {
+		min = input[i];
+	}
+	if (input[i] > max) {
+		max = input[i];
+	}
+}
```
Comment (thoanguyen2907) [inline src/assignments/6-array.js:38]: Good for the logic!

File: src/assignments/5-loop.js [hunk 5]
```diff
@@ -21,6 +21,14 @@ sum=8
 */
 // start here writing the code for program2
 
+const num = prompt("Positive number: ");
```
Comment (thoanguyen2907) [inline src/assignments/5-loop.js:24]: here need to check whether the num is a positive number or not! 
Using if else condition,  if it is not positive number give an alert !

**Review by thoanguyen2907 — COMMENTED** [review]
You did a  good job!
Understand how to write the function with or without parameter. 
Understand the code flow and have good program logic! 
👍

File: src/assignments/11-higher-order-function.js [hunk 6]
```diff
@@ -32,3 +32,17 @@ const users = [
...
+	const overThirty = users.filter(user => {
+		return user.age >= 30;
+	}).map(user => ({
+		id: user.id,
+		name: user.name,
+		country: user.country
+	}));
+	return overThirty;
+}
+console.log(usersOverThirty(users));
```
Comment (thoanguyen2907) [inline src/assignments/11-higher-order-function.js]: In this case, Anisul asked you to apply both filter or map to filter the array with the target condition. 
You can write one using filter and one using map

Comment (codecat226) [reply src/assignments/11-higher-order-function.js]: ok. now I understand the assignment thank you!

File: src/assignments/2-operator.js (hunk 2 above)
Comment (codecat226) [reply src/assignments/2-operator.js]: Now I understand why thank you!

**Comment by codecat226** [comment]
I have changed all the mistakes in the code now. Thank you for code review.

File: src/assignments/5-loop.js [hunk 7]
```diff
@@ -21,6 +21,19 @@ sum=8
...
+if (num <= 0) {
```
Comment (thoanguyen2907) [inline src/assignments/5-loop.js:25]: Good! or you can add 
```
 if(num <= 0 | isNaN(input) ) {
 }
```

File: src/assignments/11-higher-order-function.js [hunk 8]
```diff
@@ -32,3 +32,20 @@ const users = [
...
+
+const usersOverThirty = (users) => {
+	users.map((item) => {
+		if (item.age >= 30){
+			item.id,
+			item.name,
+			item.age,
+			item.country
+			console.log(item);
+		}
```
Comment (thoanguyen2907) [inline src/assignments/11-higher-order-function.js:45]: return the item here, so the new array would contains the user with age over or equal 30. 
`return item`

File: src/assignments/11-higher-order-function.js [hunk 9]
```diff
@@ -32,3 +32,20 @@ const users = [
...
+	})
+}
+
+const overThirtyFilter = (users) => users.filter(item => item.age >= 30);
```
Comment (thoanguyen2907) [inline src/assignments/11-higher-order-function.js:49]: good job!

File: src/assignments/2-operator.js [hunk 10]
```diff
@@ -14,6 +14,40 @@ output example
...
+console.log(`${num1}-${num2}=${sub}`);
+
+const multiply = num1 * num2;
+console.log(`${num1}*${num2}=${multiply}`);
+
+const divide = (num1, num2) => {
+	if (num2 == 0) {
+		console.log("undefined");
+	} else {
+		let answer = num1/num2.toFixed(2);
```
Comment (thoanguyen2907) [inline src/assignments/2-operator.js:33]: good job! always check the num2 whether bigger than 0 or not before doing divide calculation!

File: src/assignments/2-operator.js [hunk 11]
```diff
@@ -14,6 +14,40 @@ output example
...
+		return answer;
+	}
+}
+const num = divide(num1, num2);
+console.log(`${num1}/${num2}=${num}`);
+
+const remain = (num1, num2) => {
+	if (num2 == 0) {
+		console.log("undefined");
```
Comment (thoanguyen2907) [inline src/assignments/2-operator.js:42]: good!

File: src/assignments/4-switch.js [hunk 12]
```diff
@@ -8,3 +8,37 @@ vowel
...
+			console.log("vowel");
+			break;
+		case 'u':
+			console.log("vowel");
+			break;
+		case 'a':
+			console.log("vowel");
+			break;
+		default:
+			console.log("consonant");
```
Comment (thoanguyen2907) [inline src/assignments/4-switch.js:41]: good point here! necessary to have default case in switch case !

File: src/assignments/6-array.js [hunk 13]
```diff
@@ -21,3 +23,17 @@ min=1
...
+let max = input[0]; //1
+let min = input[0]; //1
+
+for (let i = 1; i < input.length; i++) {
+	if (input[i] < min) {
+		min = input[i];
+	}
+	if (input[i] > max) {
+		max = input[i];
+	}
```
Comment (thoanguyen2907) [inline src/assignments/6-array.js:37]: good logic !

**Pull request closed without merging**